streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -219,7 +223,9 @@ class CheckpointWriter(
           if (allCheckpointFiles.size > 10) {
             allCheckpointFiles.take(allCheckpointFiles.size - 10).foreach(file => {
               logInfo("Deleting " + file)
-              fs.delete(file, true)
+              if (fs.exists(file)) {
```

Comment:
Actually from the error message in Spark-9801, the only 2 checks I wanted to add were:
tempFile
backupFile

So I am fine to ignore the check for "file". Also for performance consideration.